src/java/org/apache/cassandra/replication/Shard.java

```diff
@@ -158,7 +158,25 @@ private CoordinatorLog getOrCreate(CoordinatorLogId logId)
     private CoordinatorLog getOrCreate(long logId)
     {
         CoordinatorLog log = logs.get(logId);
-        return log != null
-             ? log : logs.computeIfAbsent(logId, ignore -> CoordinatorLog.create(localHostId, new CoordinatorLogId(logId), participants));
+        if (log != null)
+            return log;
+        CoordinatorLog newLog = logs.computeIfAbsent(logId, ignore -> CoordinatorLog.create(localHostId, new CoordinatorLogId(logId), participants));
+        for (Subscriber subscriber : subscribers)
+            subscriber.onLogCreation(newLog);
+        return newLog;
+    }
+
+    private final List<Subscriber> subscribers = new ArrayList<>();
+
+    public interface Subscriber
```

Comment:
OK so if this is new then we definitely need to think through the lifecycle of creation and then expiration.

Reply:
Let's consolidate that conversation here? https://github.com/apache/cassandra/pull/4192#discussion_r2152633714